rest/config.go

```diff
@@ -1747,56 +1756,59 @@ func (sc *ServerContext) fetchDatabase(ctx context.Context, dbName string) (foun
 	return sc._fetchDatabase(ctx, dbName)
 }
 
-func (sc *ServerContext) _fetchDatabase(ctx context.Context, dbName string) (found bool, dbConfig *DatabaseConfig, err error) {
-	// loop code moved to foreachDbConfig
-	var cnf DatabaseConfig
-	callback := func(bucket string) (exit bool, err error) {
-		cas, err := sc.BootstrapContext.GetConfig(ctx, bucket, sc.Config.Bootstrap.ConfigGroupID, dbName, &cnf)
-		if err == base.ErrNotFound {
-			base.DebugfCtx(ctx, base.KeyConfig, "%q did not contain config in group %q", bucket, sc.Config.Bootstrap.ConfigGroupID)
-			return false, err
-		}
-		if err != nil {
-			base.DebugfCtx(ctx, base.KeyConfig, "unable to fetch config in group %q from bucket %q: %v", sc.Config.Bootstrap.ConfigGroupID, bucket, err)
-			return false, err
-		}
+func (sc *ServerContext) _fetchDatabaseFromBucket(ctx context.Context, bucket string, dbName string) (found bool, cnf DatabaseConfig, err error) {
 
-		if cnf.Name == "" {
-			cnf.Name = bucket
-		}
+	cas, err := sc.BootstrapContext.GetConfig(ctx, bucket, sc.Config.Bootstrap.ConfigGroupID, dbName, &cnf)
+	if err == base.ErrNotFound {
+		base.DebugfCtx(ctx, base.KeyConfig, "%q did not contain config in group %q", bucket, sc.Config.Bootstrap.ConfigGroupID)
+		return false, cnf, err
```

Comment:
This code is so wacky to me that this is basically an invalid config but we are still returning a config. 

As far as I can tell, the value of cnf is not used when `found=false` or `err != nil` value isn't used.



```mermaid
stateDiagram-v2
	handleGetDb : handleGetDb (error on not found or err)
	handleGetDbAuditConfig : handleGetDbAuditConfig (error on not found or err)
	handleGetCollectionConfigSync : handleGetCollectionConfigSync (error on not found or err)
	handleGetCollectionConfigImportFilter : handleGetCollectionConfigImportFilter (error on not found or err)
	fetchAndLoadConfigs : fetchAndLoadConfigs (cnf unused)
	_fetchAndLoadDatabase : _fetchAndLoadDatabase (cnf ignored on not found)
	
    handleGetDb --> fetchDatabase
    handleGetDbAuditConfig-->fetchDatabase
    handleGetCollectionConfigSync-->fetchDatabase
    handleGetCollectionConfingImportFilter-->fetchDatabase
    fetchDatabase-->_fetchDatabase
    fetchAndLoadConfigs-->_fetchDatabase
    _fetchAndLoadDatabase-->_fetchDatabase
```

Reply:
Isn't this fairly standard for non-pointer return values - to return an empty value when there's an error?

Reply:
I missed that this was a non pointer type, whoops.